Enonic UI: DeleteDialog #9491

skjulestad opened the pull request:
Migrated new dialog for archive / delete with new confirmation dialog as well.
<img width="929" height="899" alt="image" src="https://github.com/user-attachments/assets/33fd942e-b5bf-4004-b416-cdc8b4186cae" />

edloidas committed:
Enonic UI integration

Added `@enonic/ui` dependency.
Added Tailwind support.
Added JSX support.
Set temp version for nightly builds.
Published SNAPSHOT to repo.
Added AppBar component.
Added Sidebar component.
Added FilterPanel component.
Added BrowseToolbar component.
Added PreviewToolbar component.
Added ConfirmationDialog component.
Added ProjectSelectionDialog component.

alansemenov committed:
Disable Selenium tests

ashklianko committed:
Enonic UI: replace ContentTreeGrid #9384
- Initial ContentTreeList
- Integrate with filtering
- Expand/collapse after filtering
- Classes for Dark mode
- Integrate with Selection Toolbar (TreeListToolbar)

Enonic UI: 'Highlight' selection mode for content tree #9397

skjulestad committed:
Enonic UI: Replace existing ConfirmationDialog #9319

reisfmb committed:
Enonic UI: TreeToolbar #9390 (#9399)

edloidas committed:
Enonic UI: ThemeSwitcher #9415

Added ThemeSwitcher button.
Fixed disabled state set incorrectly on children instead of Toolbar items, which lead to incorrect navigation "skips".
Fixed cog icon of the settings.

reisfmb committed:
Enonic UI: Update TreeToolbar to match newest design #9419



---------

Co-authored-by: Mikita Taukachou <edloidas@gmail.com>

edloidas committed:
Enonic UI: Create a new PublishDialog component #9431

edloidas committed:
Enonic UI: Create a new PublishDialog component #9431

Updated agents context.

ashklianko committed:
Enonic UI: Sidebar buttons appearance issues #9440

- Added a gap between Setting button and version block
- Reduced the size of the sidebar icons according to the design

edloidas committed:
Enonic UI: Update StatusBadge #9444

edloidas committed:
Enonic UI: Create a new PublishDialog component #9431

ashklianko committed:
Enonic UI: Handle server events in Content Tree Part 1 #9438

ashklianko committed:
Enonic UI: Styling issues in Filter Panel/TreeGrid/Split Panel #9435

reisfmb committed:
Details Widget + Context View for DEMO

edloidas committed:
Enonic UI: Publish dialog UI fixes #9457

ashklianko committed:
Enonic UI: Missing Dark theme metadata and CSS rules #9455

- Presetting .dark class on the root html tag if needed
- Setting root style in an inline block

This approach prevents flickering while waiting for style scripts loaded

ashklianko committed:
Enonic UI: Focus Search input when opening filter panel #9456

edloidas committed:
Add Lib Admin UI ui2 styles to scan

Add node_modules/@enonic/lib-admin-ui/ui2 to be scanned by Tailwind to add all necessary styles to the build.

edloidas committed:
Upgrade Enonic UI dependency

ashklianko committed:
Enonic UI: Fix content tree icons in Dark mode #9461

- For ContentIcon.tsx: 'dark:invert-100 dark:brightness-75 dark:contrast-125'
- Increased contact point for selection checkbox

ashklianko committed:
Enonic UI: Toggling Context panel reveals white background in dark mode #9465

ashklianko committed:
Enonic UI: LoadMask styling #9467

edloidas committed:
Enonic UI: Missing Dark theme metadata and CSS rules #9455

Added meta "color-scheme" tag and it's update.
Added default background color.

ashklianko committed:
Enonic UI: Set the CS version to 6.0.0 for the demo #9463

edloidas committed:
Upgrade dependencies

ashklianko committed:
Enonic UI: Set uniquie ids for checkboxes #9471

- Also change the text of TreeListToolbar's checkbox label back to original

edloidas committed:
Enonic UI: Add support for server events #9473

edloidas committed:
Enonic UI: Add Show excluded toggle to Publish dialog #9475

Added "Shw excluded" toggle.
Removed legacy outline from inputs.

ashklianko committed:
Enonic UI: Version color

- Version color is updated according to the design
- Opening a content for edit on dbl click

edloidas committed:
Enonic UI: Disable Show excluded button #9483

Disabled and reset Show excluded button in Publish dialog, if nothing to show or hide.
Removed ProjectIcon margin, compensating it in ProjectItem directly.
Fixed that img icon does not receive classes from outside.

skjulestad committed:
Enonic UI: Replace export content ConfirmationDialog #9488

Replace export confirmation dialog when exporting content.

skjulestad committed:
Enonic UI: DeleteDialog #9491

Migrated new dialog for archive / delete with new confirmation dialog as well.

edloidas commented on modules/lib/src/main/resources/assets/js/v6/features/shared/dialogs/delete/DeleteDialog.tsx:
```diff
@@ -0,0 +1,166 @@
+                        />
+
+                        <Dialog.Body className="flex flex-col gap-y-10 py-5">
+                            <div className="flex flex-col gap-y-2.5">
```
No need for this div wrapper.

edloidas commented on modules/lib/src/main/resources/assets/js/v6/features/shared/dialogs/delete/DeleteDialog.tsx:
```diff
@@ -0,0 +1,166 @@
+                    setConfirmOpen(false);
+                    setConfirmAction('delete');
+                }}
+            />
```
You add DeleteDialog to the root of the app and always render it. But confirmation should not be always present in the DOM. Render it only when you require. Might not even need a separate Portal and overlay for it.

edloidas replied:
So you have single Root, single "open" and focus. But you switch content depending on the view.

edloidas replied:
I'd recommend decoupling and moving Content into separate files. Confirmations must not completely depend on the global state of the dialog or what state was set on button click. We can have two separate confirmations for Delete and Publish that will just receive the list of IDs and perform separate requests. And those requests must not depend on if the dialog is shown/loading/failed/empty or not, they are not those kinds of requests. You are already on confirmation step, all it takes is to confirm action

edloidas commented on modules/app/src/main/resources/assets/js/main.ts:
```diff
@@ -358,19 +359,11 @@ async function startApplication() {
-            .setYesCallback(event.getYesCallback())
-            .setNoCallback(event.getNoCallback())
-            .open();
+        openDeleteDialog(event.getModels(), {
+            onYes: event.getYesCallback(),
+            onNo: event.getNoCallback(),
+        });
```
1. We should not use callbacks anymore. If we need to do something on dialog visibility change, we'd rather subscribe to store.
2. I don't actually see that the yes/no callbacks are used or set anywhere. You can just drop their support.

edloidas commented on modules/lib/src/main/resources/assets/js/v6/features/shared/dialogs/delete/DeleteDialog.tsx:
```diff
@@ -0,0 +1,166 @@
+    const hasSite = useStore($deleteHasSite);
+    const inboundIds = useStore($deleteInboundIds);
+    const inboundSet = useMemo(() => new Set(inboundIds), [inboundIds]);
+    const isInbound = (id: string) => inboundSet.has(id);
```
Can pass content directly and do `item.getContentId().toString()` inside function.

edloidas commented on modules/lib/src/main/resources/assets/js/v6/features/shared/dialogs/delete/DeleteDialog.tsx:
```diff
@@ -0,0 +1,166 @@
+                            </div>
+
+                            <div className={cn('flex flex-col gap-y-7.5', dependants.length === 0 && 'hidden')}>
+                                <Separator className="" label={dependantsLabel} />
```
Add `pr-1` as it's too close to scroll

edloidas commented on modules/lib/src/main/resources/assets/js/v6/features/store/dialogs/deleteDialog.store.ts:
```diff
@@ -0,0 +1,364 @@
+            const message = total > 1
+                ? i18n('dialog.archive.success.multiple', total)
+                : i18n('dialog.archive.success.single', state.pendingPrimaryName ?? '');
+            NotifyManager.get().showSuccess(message);
```
You can call `showSuccess` as we already have those utilities, and planning to move from `NotifyManager`.

edloidas commented on modules/lib/src/main/resources/assets/js/v6/features/store/dialogs/deleteDialog.store.ts:
```diff
@@ -0,0 +1,364 @@
+    pendingPrimaryName?: string;
+    yesCallback?: () => void;
+    noCallback?: () => void;
+};
```
Store is too complex. You are mixing base properties with some properties to depend your actions and responses on. Functions that you use should be pure and do not depend on external state, only on the arguments.

edloidas commented on modules/lib/src/main/resources/assets/js/v6/features/store/dialogs/deleteDialog.store.ts:
```diff
@@ -0,0 +1,364 @@
+// * Subscriptions
+//
+
+$deleteDialog.subscribe(({open}) => {
+    if (open && !wasOpen) {
+        void reloadDeleteDialogData();
+    } else if (!open && wasOpen) {
+        instanceId += 1;
+    }
+    wasOpen = open;
+});
```
Why do we need to increment instanceId here?
Also, you can get the previous store state from second argument of subscribe handler and check if the dialog was opened.

edloidas commented on modules/lib/src/main/resources/assets/js/v6/features/store/dialogs/deleteDialog.store.ts:
```diff
@@ -0,0 +1,364 @@
+// * Completion handling
+//
+
+const handleCompletionEvent = (changeItems: ContentServerChangeItem[], kind: DeleteAction): void => {
```
I think that notifying about successful deletion or archive must not depend on server events.
Basically, you press Archive -> send request -> show message on success or error based on data you used to send request.
This handler should probably be used to reload the dialog.

edloidas commented on modules/lib/src/main/resources/assets/js/v6/features/shared/dialogs/delete/DeleteDialog.tsx:
```diff
@@ -0,0 +1,166 @@
+                            }}
+                        />
+
+                        <Dialog.Body className="flex flex-col gap-y-10 py-5">
```
Just like in publish dialog, body is incorrectly become scrolled, while having y-axis padding. This leads to the problem, when padding is also scrolled and space between the body and other elements become smaller. Add another scrollable wrapper to keep those paddings intact.

<img width="673" height="519" alt="Image" src="https://github.com/user-attachments/assets/4d13c645-71cf-4c24-b39b-a0dcc005ccbc" />

edloidas commented on modules/lib/src/main/resources/assets/js/v6/features/shared/items/ContentItem.tsx:
```diff
@@ -43,8 +48,8 @@ export const ContentItem = ({
             <ListItem.Content>
-                <Button onClick={onClick} className="block flex-1 w-[calc(100%+10px)] h-8 -mx-1.25 -my-1 px-1.25 py-1">
-                    <ListItem.DefaultContent label={label} icon={Icon} />
+                <Button onClick={onClick} className={cn('block flex-1 w-full h-8 -mx-1.25 -my-1 px-1.25 py-1', mainItem && "h-13")}>
```
The calculated width was actually used to visually enlarge the button, while keeping the content in place: we increase its size by 10px, then center the content with padding, and add negative margin to keep the content in place inside its container.
For larger content, the horizontal border should be larger. Try to apply those in the browser:
```css
margin-inline: -10px;
padding-inline: 10px;
width: calc(100% + 20px);
```

edloidas replied:
Also, would be nice to have the link active area of the same height, as the button itself, matching the row height. Right now it's not consistent.

<img width="647" height="157" alt="Image" src="https://github.com/user-attachments/assets/31b8b764-10a0-4783-820d-6942e1eb3f3f" />

skjulestad replied:
I will set it back to 100% + 10px as 20px results in overlap:
<img width="933" height="885" alt="image" src="https://github.com/user-attachments/assets/827ad813-b075-49c6-9cb2-0cf1ec7b9a26" />

Height was set for sub items but is now adjusted based on if it is mainItem or not:

<img width="933" height="885" alt="image" src="https://github.com/user-attachments/assets/9064b82a-6457-4204-8b61-95f29cfde619" />
<img width="933" height="484" alt="image" src="https://github.com/user-attachments/assets/426f621b-a665-49f1-9431-e1028611b126" />

Also gave some more space for scrollbar:
<img width="933" height="888" alt="image" src="https://github.com/user-attachments/assets/8be2811d-91bf-444c-b2ed-34b692beb135" />

reisfmb committed:
Enonic UI: Details Widget refactoring #9493

Added new translation keys
Added stores and utility hooks
Added shared icon components and utility file for details widget
Updated all details widget sub components
Added details widget template section
Deprecated legacy details widget item views
Updated context view and widgets dropdown
Improved detailsWidgets store and fixed content section status
Minor improvement on detailsWidgetElement usage

---------

Co-authored-by: Mikita Taukachou <edloidas@gmail.com>

edloidas commented:
We still need to update how we render the dialogs.
Currently, you show new dialog as a separate dialog. This is not quite correct. It makes overlay to re-render every time and complicates the data management:
- Return from `DeleteDialog` a single `<Dialog.Root>` node from top level.
- Add new local type called `type View = 'main' | 'confirmation' | 'progress'`
- Reset that `view` state onClose, no need to keep it in global store.
- Depending on the current view you should render either Delete content, or Confirmation content.
- You'll have to rework delete confirmation preset to return Content only, not the whole dialog. Footer of the confirmation content must also be adjusted, so that the buttons won't trigger close.

I updated the OnlineStatus. Keep it as is for now, we will refactor status elements in a separate branch.

skjulestad committed:
Enonic UI: DeleteDialog #9491

Migrated new dialog for archive / delete with new confirmation dialog as well.
Fix button design.
Refactored after review.
